faq.md

```diff
@@ -103,3 +103,31 @@ than a 1000 symbols and you would like to get through all of them, please pass
 
 The same applies to the Remote Index Service but we are not respecting
 `--limit-references=0` on the server side to prevent DDoS attacks.
+
+## How do I fix errors I get when opening headers outside of my project directory?
```

Comment:
This contains really useful information, but I think it is a little bit too long for FAQ. So what about dropping the first paragraph completely and moving the second one into https://clangd.llvm.org/design/compile-commands#compile-command-problems and then just saying something like

```
Clangd might fail to find suitable compile flags for sources outside of your project directory (e.g. third party dependencies installed elsewhere, for more details see https://clangd.llvm.org/design/compile-commands).
You can work around such problems by forcing clangd to look at compilation database in a particular directory with ....
```

Reply:
Thanks, I didn't know about this page! Update the patch as suggested.

One question: a lot of the material on the "Compile commands" page seems useful for **users**, but it's in a section ("Design") where some other material (e.g. the "Code walkthrough" page) has **contributors** as the intended audience.

Do you think it might be worth revising this (in a future change) so the material relevant to users is in a place they're more likely to look?

Reply:
> One question: a lot of the material on the "Compile commands" page seems useful for users, but it's in a section ("Design") where some other material (e.g. the "Code walkthrough" page) has contributors as the intended audience.

The main idea for that document was to describe the imprortance of compile flags for clangd and how they are figured out in certain cases, which talks about lots of details. Hence we assumed it won't be useful to the general user. Do you have any specific sections in mind?

Reply:
* The [Compile command vs. clangd flags](https://clangd.llvm.org/design/compile-commands#compile-command-vs-clangd-flags) section seems like it clears up a confusion more likely to be experienced by users new to the project.
 * The [Compile command problems](https://clangd.llvm.org/design/compile-commands#compile-command-problems) section seems useful as a troubleshooting guide for users.